prog/dftb+/lib_dftb/etemp.F90

```diff
@@ -449,6 +481,24 @@ subroutine electronFill(Eband, filling, TS, E0, Ef, eigenvals, kT, distrib, kWei
       TS = TS * kT
       E0(:) = (real(MPorder + 1,dp) * (Eband - TS) + Eband) / real(MPorder + 2, dp)
     else
+! TI-DFTB Non-Aufbau filling routine
+      if (tNonAufbau) then
+        allocate(tmpMtx(size(eigenvals, dim=1),kpts,size(eigenvals, dim=3)))
+        tmpMtx=eigenvals
+        do iSpin = 1, size(eigenvals, dim=3)
+          do i = 1, kpts
+            if (iDet == 1 .and. tSpinPurify .and. iS==1) then
+              eigenvals(int(nEl(iS)) + 1, i, iSpin)=eigenvals(int(nEl(iS)), i, iSpin)
```

Comment:
Shouldn't eigenvals(int(nEl(iS)), i, iSpin) then be over-written with the original value of eigenvals(int(nEl(iS)) + 1, i, iSpin) ? Also, any reason for using int (this would behave like ceiling() as nEl >= 0, which might be clearer) or do you actually need nint() which would be robust to numerical noise in nEl?